streamer/src/nonblocking/quic.rs

```diff
@@ -519,10 +558,72 @@ async fn setup_connection(
     }
 }
 
+async fn packet_batch_sender(
+    packet_sender: Sender<PacketBatch>,
+    packet_receiver: AsyncReceiver<PacketAccumulator>,
+    exit: Arc<AtomicBool>,
+    stats: Arc<StreamStats>,
+) {
+    trace!("enter packet_batch_sender");
+    loop {
+        let mut packet_batch = PacketBatch::with_capacity(PACKET_BATCH_SIZE);
+
+        stats
+            .total_packets_allocated
+            .fetch_add(PACKET_BATCH_SIZE, Ordering::Relaxed);
+
+        let mut last_sent = Instant::now();
+        loop {
+            if exit.load(Ordering::Relaxed) {
+                return;
+            }
+            let elapsed = last_sent.elapsed();
+            if packet_batch.len() >= PACKET_BATCH_SIZE
+                || (!packet_batch.is_empty() && elapsed.as_millis() >= COALESCE_MS)
+            {
+                let len = packet_batch.len();
+                if let Err(e) = packet_sender.send(packet_batch) {
+                    stats
+                        .total_packet_batch_send_err
+                        .fetch_add(1, Ordering::Relaxed);
+                    trace!("Send error: {}", e);
+                } else {
+                    stats
+                        .total_packet_batches_sent
+                        .fetch_add(1, Ordering::Relaxed);
```

Comment:
should not have the count fetch_add(len)? Are we counting the batches or send? The name indicates the former

Reply:
Yup, this, I think, counts the number of packet batches sent. So we increment it by 1 every time we send a packet batch, rather than the length of the packet batch.